src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -223,6 +229,7 @@ async def mypy_typecheck_partition(
         "PEX_EXTRA_SYS_PATH": ":".join(all_used_source_roots),
         "MYPYPATH": ":".join(all_used_source_roots),
     }
+    cache_dir = f".cache/mypy_cache/{hex(abs(hash(build_root.path)))[2:]}"
```

Comment:
Python's `hash` function might not be strong enough for this usecase: would use `hashlib` `sha256`.

Reply:
Python's `hash()` function is not deterministic across processes, so this wouldn't even have been correct IIUC.